ADAL/src/public/ADTelemetry.h

```diff
@@ -34,28 +34,9 @@
 
 /*!
     Callback function that will be called by ADAL when telemetry events are flushed.
-    @param  event        An event is represented by an array of ADTelemetryProperty.
+    @param  event        An event is represented by a dictionary of key-value properties.
  */
-- (void)dispatchEvent:(NSArray*)event;
-
-@end
-
-/*!
-    @class ADTelemetryProperty
- 
-    Each telemetry event dispatched to the above callback function is represented by an array of ADTelemetryProperty.
-    It includes property name and property value.
- */
-@interface ADTelemetryProperty : NSObject
-{
-    NSString* _name;
-    NSString* _value;
-}
-
-@property (readonly) NSString* name;
-@property (readonly) NSString* value;
-
-- (id)initWithName:(NSString*)name value:(NSString*)value;
+- (void)dispatchEvent:(nonnull NSDictionary<NSString*, NSString*> *)event;
```

Comment:
Naming suggestion: aren't these actually properties of the event? It'd make the argument a little clearer to call it `properties` rather than `event`.

alternatively, why not just pass the `ADTelemetryEventInterface` object to the delegate? The interface is a little confusing because you expect to receive an "event", but instead you get a bunch of properties that describe the event.